cpu/nrf5x_common/periph/uart.c

```diff
@@ -115,31 +116,29 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 
 #ifdef CPU_MODEL_NRF52840XXAA
     /* enable HW-flow control if defined */
-    if (UART_HWFLOWCTRL) {
+ #ifdef MODULE_PERIPH_UART_HW_FC
         /* set pin mode for RTS and CTS pins */
+        if (UART_PIN_RTS != GPIO_UNDEF || UART_PIN_CTS != GPIO_UNDEF) {
+            gpio_init(UART_PIN_RTS, GPIO_OUT);
+            gpio_init(UART_PIN_CTS, GPIO_IN);
+            /* configure RTS and CTS pins to use */
+            dev(uart)->PSEL.RTS = UART_PIN_RTS;
+            dev(uart)->PSEL.CTS = UART_PIN_CTS;
+            REG_CONFIG |= UART_CONFIG_HWFC_Msk; /* enable HW flow control */
+        }
+#endif
+#else
+#ifdef MODULE_PERIPH_UART_HW_FC
+    /* set pin mode for RTS and CTS pins */
+    if (UART_PIN_RTS != GPIO_UNDEF || UART_PIN_CTS != GPIO_UNDEF) {
         gpio_init(UART_PIN_RTS, GPIO_OUT);
         gpio_init(UART_PIN_CTS, GPIO_IN);
         /* configure RTS and CTS pins to use */
-        dev(uart)->PSEL.RTS = UART_PIN_RTS;
-        dev(uart)->PSEL.CTS = UART_PIN_CTS;
-        REG_CONFIG |= UART_CONFIG_HWFC_Msk; /* enable HW flow control */
-    } else {
```

Comment:
This else is gone, same for non nrf52840 block

Reply:
Should be good.